### PR TITLE
Make plugin autoclonetype always on

### PR DESCRIPTION
**Pull request by jackkoenig**
Follow on to https://github.com/chipsalliance/chisel3/pull/1804, make the compiler plugin autoclonetype be always on and thus used in chisel3 itself.

### Contributor Checklist

- [NA] Did you add Scaladoc to every public function/method?
- [x] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [x] Did you state the API impact?
- [x] Did you specify the code generation impact?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

- performance improvement 
 - new feature/API  

#### API Impact

As described in the release notes in https://github.com/chipsalliance/chisel3/pull/1804 and reflected in the test change, this will cause issues for any Bundle that inherits from a Bundle in `chisel3.util` if the code is not also compiled using the compiler plugin. Thus, this change can only occur on the major version bump.

#### Desired Merge Strategy

- Squash

#### Release Notes

Always generate `cloneType` methods for Bundles using the compiler plugin. This causes a binary and source incompatible change for any Chisel code that does not use the compiler plugin but extends a `Bundle` in other code that does. All Chisel users should use the compiler plugin.

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (3.2.x, 3.3.x, 3.4.0, 3.5.0) ?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you mark as `Please Merge`?

### CODE REVIEW

File: plugin/src/main/scala-2.12/chisel3/internal/plugin/ChiselPlugin.scala [hunk 1]
```diff
@@ -24,7 +25,8 @@ class ChiselPlugin(val global: Global) extends Plugin {
   override def init(options: List[String], error: String => Unit): Boolean = {
     for (option <- options) {
       if (option == arguments.useBundlePluginOpt) {
-        arguments.useBundlePlugin = true
+        val msg = s"'${arguments.useBundlePluginFullOpt}' is now default behavior, you can stop using the scalacOption."
```
Comment (azidar) [inline plugin/src/main/scala-2.12/chisel3/internal/plugin/ChiselPlugin.scala:28]: Can you write the actual scalacOption here in the message?

File: src/test/scala/chiselTests/AutoClonetypeSpec.scala [hunk 2]
```diff
@@ -265,19 +265,18 @@ class AutoClonetypeSpec extends ChiselFlatSpec with Utils {
...
-      val io = IO(new InheritingBundle)
-      io.deq <> io.enq
-      io.count := 0.U
-      io.error := true.B
-    }
-    elaborate(new MyModule)
-  }
-
   // New tests from the plugin
   if (usingPlugin) {
```
Comment (azidar) [inline src/test/scala/chiselTests/AutoClonetypeSpec.scala:269]: Do we need this anymore?

**Review by azidar — CHANGES_REQUESTED** [review]

File: plugin/src/main/scala-2.12/chisel3/internal/plugin/ChiselPlugin.scala (hunk 1 above)
Comment (jackkoenig) [reply plugin/src/main/scala-2.12/chisel3/internal/plugin/ChiselPlugin.scala:28]: That's what the string interpolator is doing!

File: src/test/scala/chiselTests/AutoClonetypeSpec.scala (hunk 2 above)
Comment (jackkoenig) [reply src/test/scala/chiselTests/AutoClonetypeSpec.scala:269]: Yes, the `noPluginTests` compile and run this same file https://github.com/chipsalliance/chisel3/blob/03ec2156988ddef35272365fb86415511eafb4e2/build.sbt#L206

We should consider making the plugin mandatory but that's not the purview of this PR

**Review by azidar — APPROVED** [review]

**Pull request merged**